unix/Makefile

```diff
@@ -61,6 +61,7 @@ endif
 # if necessary override the value of 'CC' set in py/mkenv.mk
 ifeq ($(UNAME_S),Darwin)
 CC = clang
+CFLAGS += -Wno-shift-negative-value
```

Comment:
This is the error message:

```c
../py/objlist.c:272:78: error: shifting a negative signed value is undefined [-Werror,-Wshift-negative-value]
    mp_uint_t index = mp_get_index(self->base.type, self->len, n_args == 1 ? MP_OBJ_NEW_SMALL_INT(-1) : args[1], false);
                                                                             ^~~~~~~~~~~~~~~~~~~~~~~~
../py/obj.h:86:78: note: expanded from macro 'MP_OBJ_NEW_SMALL_INT'
#define MP_OBJ_NEW_SMALL_INT(small_int) ((mp_obj_t)((((mp_int_t)(small_int)) << 1) | 1))
                                                     ~~~~~~~~~~~~~~~~~~~~~~~ ^
1 error generated.
```

The other way how to prevent the error on Mac is to cast the `small_int` to unsigned ``mp_uint_t``:
```c
#define MP_OBJ_NEW_SMALL_INT(small_int) ((mp_obj_t)((((mp_uint_t)(small_int)) << 1) | 1))
```
The MP_OBJ_SMALL_INT_VALUE macro converts it back to ``mp_int_t`` correctly.